lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -141,6 +141,7 @@ def inline_metadata(catalog, checksum_type)
 
           # TODO: Conditionally copy owner, group, and mode when we can check if permissions are set
           child_resource = Puppet::Resource.new(:file, File.join(file[:path], meta.relative_path))
+          child_resource[:ensure] = meta.ftype
```

Comment:
we'll need to remove the `to_ral` call later on, and call `Puppet::FileServing::Metadata.indirection.{find,search}` directly, instead of creating a `Puppet::Type::File` instance.